thoth/storages/graph/postgres.py

```diff
@@ -3846,26 +3847,48 @@ def create_python_cve_record(
         version_range: str,
         advisory: str,
         cve: str = None,
-    ) -> Tuple[CVE, bool]:
+    ) -> bool:
         """Store information about a CVE in the graph database for the given Python package."""
         package_name = self.normalize_python_package_name(package_name)
         package_version = self.normalize_python_package_version(package_version)
         with self._session_scope() as session, session.begin(subtransactions=True):
-            cve, _ = CVE.get_or_create(
-                session, cve_id=record_id, version_range=version_range, advisory=advisory, cve_name=cve
-            )
-            index = self._get_or_create_python_package_index(session, index_url, only_if_enabled=False)
-            entity, _ = PythonPackageVersionEntity.get_or_create(
-                session,
-                package_name=package_name,
-                package_version=package_version,
-                python_package_index_id=index.id,
-            )
-            _, existed = HasVulnerability.get_or_create(
-                session, cve_id=cve.id, python_package_version_entity_id=entity.id
-            )
+            if session.query(exists().where(CVE.cve_id == record_id)).scalar():
+                return True
+            else:
+                kwargs = {
+                    "advisory": advisory,
+                    "cve_name": cve,
+                    "cve_id": record_id,
+                    "version_range": version_range,
+                    "aggregated_at": datetime.now(),
+                }
+                try:
+                    session.begin_nested()
+                    cve_instance = CVE(**kwargs)
+                    session.add(cve_instance)
+                    session.commit()
+                except IntegrityError as exc:
```

Comment:
This is already handled by the context manager. You can simply do:

```python
                    cve_instance = CVE(
                        # params...
                    )
                    session.add(cve_instance)
```

The context manager will make sure everything is committed or rolled back on any failure.

Reply:
Wow, thanks, I was figuring that out, :smile: